src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2152,18 +2317,23 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
 
             MethodDesc targetMethod = methodAfterConstraintResolution;
 
-            // TODO: Support Generics
+            //
+            // Initialize callee context used for inlining and instantiation arguments
+            //
+
+
             if (targetMethod.HasInstantiation)
             {
                 pResult.contextHandle = contextFromMethod(targetMethod);
+                pResult.exactContextNeedsRuntimeLookup = targetMethod.IsSharedByGenericInstantiations;
             }
             else
             {
-                pResult.contextHandle = contextFromType(targetMethod.OwningType);
+                // TODO: There was an "if (!exactType.IsTypeDesc())" on the CoreCLR side here 
```

Comment:
I think it is in CoreCLR to protect against array methods - they have TypeDesc type in CoreCLR that is different from their MethodTable.

It should be safe to remove this special case here.